src/plots/cartesian/set_convert.js

```diff
@@ -448,11 +448,10 @@ module.exports = function setConvert(ax, fullLayout) {
 
     ax.isPtWithinRange = function(d, calendar) {
         var coord = ax.c2l(d[axLetter], null, calendar);
+        var r0 = ax.r2l(ax.range[0]);
+        var r1 = ax.r2l(ax.range[1]);
 
-        return (
-            coord >= ax.r2l(ax.range[0]) &&
-            coord <= ax.r2l(ax.range[1])
-        );
+        return Math.min(r0, r1) <= coord && coord <= Math.max(r0, r1);
```

Comment:
Perhaps:

```js
var r0 = ax.r2l(ax.range[0]);
var r1 = ax.r2l(ax.range[1]);

if(r0 < r1) {
  return coord >= r0 && coords <= r1;
} else {
  // reversed axis case
  return coord <= r0 && coords >= r1;
}
```

would perform (one comparison instead of `Math.min` + `Math.max`) and arguably more readable.

Reply:
Thanks, done. I will try to figure out how to recreate this pull request against the main repo.